### PR TITLE
Transition Beluga to Noble with Rolling

### PR DESCRIPTION
**Pull request by hidmic**
### Proposed changes

ROS 2 Rolling Ridley has moved on from Ubuntu Jammy. This patch updates Beluga packages to follow Rolling to Ubuntu Noble.

#### Type of change

- [x] 🐛 Bugfix (change which fixes an issue)
- [ ] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### Additional comments

The build is not quite fixed yet because not all our dependencies are available yet (or even build) for Rolling on Noble.

### CODE REVIEW

File: beluga/test/beluga/sensor/data/test_occupancy_grid.cpp [hunk 1]
```diff
@@ -83,8 +83,13 @@ TEST(OccupancyGrid2, AllFreeCells) {
 TEST(OccupancyGrid2, ObstacleData) {
   const auto grid = StaticOccupancyGrid<5, 2>{{false, false, false, false, true, false, false, false, true, false}};
 
+// See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111118
```
Comment (hidmic) [inline beluga/test/beluga/sensor/data/test_occupancy_grid.cpp:86]: That `std::vector<bool>` "optimization" will haunt C++ programmers forever.

File: docker/images/rolling/Dockerfile [hunk 2]
```diff
@@ -52,6 +52,7 @@ RUN apt-get update \
     tmux \
   && rm -rf /var/lib/apt/lists/*
 
+ENV PIP_BREAK_SYSTEM_PACKAGES 1
```
Comment (hidmic) [inline docker/images/rolling/Dockerfile:55]: By default, `pip` now won't allow system-wide installations. I wonder how the ROS folks will deal with this in `rosdep`.

File: docker/images/rolling/Dockerfile [hunk 3]
```diff
@@ -98,7 +99,7 @@ RUN sudo apt-get update \
   && sudo apt-get -y dist-upgrade \
   && . /opt/ros/rolling/setup.sh \
   && rosdep update \
-  && rosdep install -i -y --from-path src --skip-keys 'slam_toolbox' \
+  && rosdep install -i -y --from-path src --skip-keys 'turtlebot3_gazebo gazebo_ros_pkgs slam_toolbox' \
```
Comment (hidmic) [inline docker/images/rolling/Dockerfile:102]: These packages have not yet been released. Also `flatland` and `nav2_map_server` still fail to build.

**Review by nahueespinosa — APPROVED** [review]
LGTM!

File: beluga/test/beluga/sensor/data/test_occupancy_grid.cpp [hunk 4]
```diff
@@ -83,8 +83,13 @@ TEST(OccupancyGrid2, AllFreeCells) {
...
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
+#pragma GCC diagnostic ignored "-Wstringop-overflow"
   // Data and obstacle data are equivalent in this case
   ASSERT_THAT(grid.obstacle_data() | ranges::to<std::vector>, testing::Pointwise(testing::Eq(), grid.data()));
+#pragma GCC diagnost pop
```
Comment (nahueespinosa) [inline beluga/test/beluga/sensor/data/test_occupancy_grid.cpp:92]:
```suggestion
#pragma GCC diagnostic pop
```

Comment (hidmic) [reply beluga/test/beluga/sensor/data/test_occupancy_grid.cpp]: How this didn't come up locally is beyond me :man_facepalming:

**Comment by nahueespinosa** [comment]
> @nahueespinosa is there a threshold we can set for codecov not to complain about 0.0x% changes in test coverage? 

Yes, let me put something up later today.

**Pull request merged**